tmt/steps/prepare/artifact/providers/__init__.py

```diff
@@ -161,10 +176,13 @@ def _filter_artifacts(self, exclude_patterns: list[Pattern[str]]) -> Iterator[Ar
         """
         Filter artifacts based on exclude patterns.
 
-        :param exclude_patterns: The patterns to exclude.
-        :yields: Artifacts that do not match any of the exclude patterns.
+        :param exclude_patterns: artifact whose name matches any of
+            these patterns would be skipped.
+        :yields: artifacts that satisfy the filtering.
         """
 
         for artifact in self.list_artifacts():
-            if not any(pattern.search(artifact.id) for pattern in exclude_patterns):
+            if not exclude_patterns or not any(
```

Comment:
The separaye `if not exclude_patterns` is not needed

Reply:
Dropped in 4c5e1254.